packages/@lwc/engine-core/src/3rdparty/snabbdom/snabbdom.ts

```diff
@@ -148,7 +149,11 @@ export function updateDynamicChildren(parentElm: Node, oldCh: VNodes, newCh: VNo
     }
     if (oldStartIdx <= oldEndIdx || newStartIdx <= newEndIdx) {
         if (oldStartIdx > oldEndIdx) {
-            const n = newCh[newEndIdx + 1];
+            let i = newEndIdx;
```

Comment:
i need to give it more thought, but this should do it; what do you think?

Reply:
No no, lets roll back this change... it makes me very nervous to change this algo. The other change that I pushed should be enough. Isn't it?

Reply:
@caridy i updated the pr with some new tests cases. those test cases does not involve slots.